tests/xbot/common/logic/LatchTest.java

```diff
@@ -0,0 +1,133 @@
+package xbot.common.logic;
+
+import org.junit.Test;
+
+import java.util.Observable;
+import java.util.Observer;
+
+import static junit.framework.TestCase.assertEquals;
+
+public class LatchTest {
+
+    @Test
+    public void testRisingEdge() {
+        Latch latch = new Latch(false, Latch.EdgeType.RisingEdge);
+        LatchTestObserver latchTestObserver = new LatchTestObserver();
+        latch.addObserver(latchTestObserver);
+        assertEquals(latchTestObserver.getLastUpdateEdgeType(), null);
+        latch.setValue(true);
+        assertEquals(latchTestObserver.getLastUpdateEdgeType(), Latch.EdgeType.RisingEdge);
```

Comment:
Aren't these method parameters supposed to be `expected, actual`? I can't remember exactly.

Reply:
I don't write much test so I forgot